Formula/leakcanary-shark.rb

```diff
@@ -0,0 +1,22 @@
+class LeakcanaryShark < Formula
+  desc "CLI Java memory leak explorer for LeakCanary"
+  homepage "https://square.github.io/leakcanary/shark/"
+  url "https://github.com/square/leakcanary/releases/download/v2.2/shark-cli-2.2.zip"
+  sha256 "4d8aaeb68e2783217267109f5d0c7bcafc78607b54b248e3f3214c07777fd278"
+
+  bottle :unneeded
+
+  depends_on :java => "1.8+"
+
+  def install
+    # Remove Windows scripts
+    rm_rf Dir["bin/*.bat"]
+
+    libexec.install Dir["*"]
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "Usage: shark-cli", shell_output("#{bin}/shark-cli").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.